services/sso/src/test/java/software.amazon.awssdk.services.sso/auth/SsoCredentialsProviderTest.java

```diff
@@ -0,0 +1,275 @@
+                                "sso_region=us-east-1\n" +
+                                "sso_start_url=" + START_URL + "\n";
+        missingProfileContentTestHelper(profileContent,"sso_role_name");
+    }
+
+    @Test
+    public void createSSOCredentialsProvider_missingStartUrl_throwSdkClientException() {
+        String profileContent = "[profile foo]\n" +
+                                "sso_account_id=012345678901\n" +
+                                "sso_region=us-east-1\n" +
+                                "sso_role_name=SampleRole\n";
+        missingProfileContentTestHelper(profileContent,"sso_start_url");
+    }
+
+    private void missingProfileContentTestHelper(String profileContent, String missingContent) {
+        SsoClient ssoClient = mock(SsoClient.class);
+        ProfileFile profiles = ProfileFile.builder()
+                                          .content(new StringInputStream(profileContent))
+                                          .type(ProfileFile.Type.CONFIGURATION)
+                                          .build();
+
+
+        provider = SsoCredentialsProvider.builder()
+                                         .profileFile(profiles)
+                                         .profileName("foo")
+                                         .ssoClient(ssoClient)
+                                         .build();
+        exception.expect(SdkClientException.class);
+        exception.expectMessage("This profile is invalid, " + missingContent + " is missing.");
+        provider.resolveCredentials();
+        provider.close();
+    }
+
+    private static void createTestTokenFile(String tokenDirectory, String expiration) throws Exception {
+        String tokenFileContent = "{\n" +
+                                  "\"accessToken\": \"base64string\",\n" +
+                                  "\"expiresAt\": \"" + expiration + "\",\n" +
+                                  "\"region\": \"us-west-2\",\n" +
+                                  "\"startUrl\": \"https//d-abc123.awsapps.com/start\"\n" +
+                                  "}";
+        try{
+            File fakeTokenFile = new File(tokenDirectory);
+            if (fakeTokenFile.createNewFile()) {
+                FileWriter fileWriter = new FileWriter(fakeTokenFile);
+                fileWriter.write(tokenFileContent);
+                fileWriter.close();
+            }
+        } catch (IOException e) {
+            throw new Exception(String.format("Creating the test token file failed because of %s", e));
+        }
+
+    }
+
+    private static String generateCachedTokenPath(String startUrl) {
+        Validate.notNull(startUrl, "The start url shouldn't be null.");
+        byte[] startUrlBytes = startUrl.getBytes(StandardCharsets.UTF_8);
+        String encodedUrl = new String(startUrlBytes, StandardCharsets.UTF_8);
+        return "/Users/"+System.getProperty("user.name")+"/.aws/sso/cache/" +  sha1Hex(encodedUrl) + ".json";
+    }
```

Comment:
It's bad form to reproduce the logic you're testing within your tests (e.g. sha1Hex, generateCachedTokenPath), because you're likely going to write the same bug in both places and defeat the purpose of your tests. Can you just hard-code the values you are expecting here?